### PR TITLE
Feature/config

### PR DESCRIPTION
**Pull request by hack1337**
- GET and PUT configs (APPLICATION, ENVIRONMENT, ONE ENVIRONMENT, LANGUAGE, SECURITY)
- Tests

### CODE REVIEW

File: templates/services/ConfigService.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    _.forEach(scopeTypes, function(type) {
+      if (typeof params[value] === type) isValideType = true;
+    });
+    if (!isValideType) {
+      error = true;
+      return;
+    }
+
+    // Recursive to process keys settings of an object
+    if (_.include(scopeTypes, 'object') && typeof params[value] === 'object') {
```
Comment (hack1337) [inline templates/services/ConfigService.js]: _.isObject(params[value]) ?

File: templates/controllers/ConfigController.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+var Promise = require('bluebird');
+
+module.exports = {
+
+  // APPLICATION
+  getApplication: function(req, res) {
+    res.json({
+      name: sails.config.application.name,
+      description: sails.config.application.description,
+      version: sails.config.application.version,
```
Comment (Aurelsicoko) [inline templates/controllers/ConfigController.js]: Remove comma

File: templates/controllers/ConfigController.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    ConfigService.rewriteConfigFile(path, newConfig)
+      .then(function(data) {
+
+        scope = _.union(scope, [{
+          name: 'environments',
+          path: 'environments'
+        }]);
+
+        var params = ConfigService.responseConstructor(scope);
+
```
Comment (Aurelsicoko) [inline templates/controllers/ConfigController.js]: Useless empty line

File: templates/controllers/ConfigController.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      p3p: sails.config.p3p,
+      hsts: sails.config.hsts,
+      xssProtection: sails.config.xssProtection
+    })
+  },
+
+  updateSecurity: function(req, res) {
+    var acceptedValues = ['DENY', 'SAMEORIGIN', 'ALLOW-FROM '];
+    var path = './config/security.json';
+    var newConfig = {};
```
Comment (Aurelsicoko) [inline templates/controllers/ConfigController.js]: This var is re-initialized after ;)

File: templates/services/ConfigService.js [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  // Map each request params
+  _.map(_.keys(params), function(value) {
+    // Retrieving parameter settings
+    var scopeConfig = _.find(scope, {
+      name: value
+    });
+
+    // Type checking
+    scopeTypes = scopeConfig.type.split('/');
+    var isValideType = false;
```
Comment (Aurelsicoko) [inline templates/services/ConfigService.js]: Change var name to "isValidType"

**Comment by Aurelsicoko** [comment]
OK

File: templates/services/ConfigService.js (hunk 5 above)
Comment (hack1337) [inline templates/services/ConfigService.js]: DONE

File: templates/controllers/ConfigController.js (hunk 4 above)
Comment (hack1337) [inline templates/controllers/ConfigController.js]: DONE

File: templates/controllers/ConfigController.js (hunk 2 above)
Comment (hack1337) [inline templates/controllers/ConfigController.js]: DONE

**Pull request merged**